modules/kubernetes-addons/prometheus/main.tf

```diff
@@ -75,4 +75,52 @@ resource "helm_release" "prometheus" {
     }
   }
 
+  depends_on = [kubernetes_namespace_v1.prometheus]
+
+}
+
+resource "kubernetes_namespace_v1" "prometheus" {
+  metadata {
+    name = local.helm_config["namespace"]
+    labels = {
+      "app.kubernetes.io/managed-by" = "terraform-ssp-amazon-eks"
+    }
+  }
+}
+
+# Amazon Managed Prometheus Resources
+
+resource "aws_prometheus_workspace" "amp_workspace" {
```

Comment:
I don't think we want to provide customers with multiple places to create an amp_workspace in the framework. Either we keep it in the dedicated submodule or we move it here.

Reply:
As you know `kubernetes-addons` is an independent sub module. It can be used without any dependency on core module. Having this feature allows customer to create workspace if they don't provide any existing workspace id.

We can remove the core AMP module if that's not necessary